### PR TITLE
Add xrpld build option and Conan package test

### PR DESCRIPTION
**Pull request by thejohnfreeman**
Related to #5028. The Clio trigger from that branch needs to be merged with this one.

### CODE REVIEW

File: examples/example/src/example.cpp [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#include <cstdio>
+
+#include <xrpl/protocol/BuildInfo.h>
+
+int main(int argc, const char** argv) {
```
Comment (godexsoft) [inline examples/example/src/example.cpp:5]: nit: I thought rippled codebase preferred east-const, not that it matters ☝️

**Review by godexsoft — APPROVED** [review]
I like the `with_xrpld` change. We should start using it in Clio asap too 👍
Imo worth merging. We can add the Clio notification on top of it right here or in the other PR once this is merged.

**Comment by godexsoft** [comment]
@Bronek this is still awaiting your review 🙇

File: .github/workflows/nix.yml [hunk 2]
```diff
@@ -232,3 +232,52 @@ jobs:
...
+        run: |
+          cd examples/example
+          mkdir ${build_dir}
+          cd ${build_dir}
+          conan install .. --output-folder . \
+            --require-override ${reference} --build missing
+          cmake .. \
+            -DCMAKE_TOOLCHAIN_FILE:FILEPATH=./build/${configuration}/generators/conan_toolchain.cmake \
+            -DCMAKE_BUILD_TYPE=${configuration}
+          cmake --build .
```
Comment (Bronek) [inline .github/workflows/nix.yml:283]: I would also add a step to run the example and ` | grep -F "${version}"` from its output, but that's just me.

Also, I love the idea behind this newly added workflow.

File: examples/example/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+        'xrpl:with_xrpld': False,
+    }
+
+    requires = ['xrpl/2.2.0-rc1@jfreeman/nodestore']
+    generators = ['CMakeDeps', 'CMakeToolchain']
+
+    exports_sources = [
```
Comment (Bronek) [inline examples/example/conanfile.py:28]: Is this needed in this `example` project ?

**Review by Bronek — APPROVED** [review]
I like it, just some minor comments

File: examples/example/conanfile.py (hunk 3 above)
Comment (thejohnfreeman) [reply examples/example/conanfile.py:28]: No, it's just part of the template I use.

**Comment by thejohnfreeman** [comment]
I renamed the option to be more consistent with the other option names (especially `tests`).

**Pull request merged**